core/src/main/java/com/graphhopper/routing/subnetwork/PrepareRoutingSubnetworks.java

```diff
@@ -56,155 +68,118 @@ public PrepareRoutingSubnetworks(GraphHopperStorage ghStorage, List<PrepareJob>
+     * as subnetworks. The biggest component will never be marked as subnetwork, even when it is below this size.
      */
     public PrepareRoutingSubnetworks setMinNetworkSize(int minNetworkSize) {
         this.minNetworkSize = minNetworkSize;
         return this;
     }
 
-    public void doWork() {
+    /**
+     * Finds and marks all subnetworks according to {@link #setMinNetworkSize(int)}
+     *
+     * @return the total number of marked edges
+     */
+    public int doWork() {
         if (minNetworkSize <= 0) {
-            logger.info("Skipping subnetwork removal: prepare.min_network_size: " + minNetworkSize);
-            return;
+            logger.info("Skipping subnetwork search: prepare.min_network_size: " + minNetworkSize);
+            return 0;
         }
         StopWatch sw = new StopWatch().start();
-        logger.info("Start removing subnetworks, prepare.min_network_size: " + minNetworkSize + ", nodes: " +
+        logger.info("Start marking subnetworks, prepare.min_network_size: " + minNetworkSize + ", nodes: " +
                 Helper.nf(ghStorage.getNodes()) + ", edges: " + Helper.nf(ghStorage.getEdges()) + ", jobs: " + prepareJobs + ", " + Helper.getMemInfo());
-        for (PrepareJob job : prepareJobs) {
-            setSubnetworks(job);
-        }
-        logger.info("Finished finding and removing subnetworks for " + prepareJobs.size() + " vehicles, took: " + sw.stop().getSeconds() + "s, " + Helper.getMemInfo());
-    }
-
-    /**
-     * The biggest component is always kept regardless of its size. For edge-based routing with turn restrictions the
-     * subnetwork search has to consider the turn restrictions as well to make sure components that are not reachable
-     * due to turn restrictions are also removed.
-     *
-     * @return number of removed edges
-     */
-    int setSubnetworks(PrepareJob job) {
-        return setSubnetworks(job.name, job.weighting, job.inSubnetworkEnc);
+        int total = 0;
+        for (PrepareJob job : prepareJobs)
+            total += setSubnetworks(job.weighting, job.subnetworkEnc);
+        logger.info("Finished finding and marking subnetworks for " + prepareJobs.size() + " jobs, took: " + sw.stop().getSeconds() + "s, " + Helper.getMemInfo());
+        return total;
     }
 
-    private int setSubnetworks(String jobName, Weighting weighting, BooleanEncodedValue inSubnetworkEnc) {
+    private int setSubnetworks(Weighting weighting, BooleanEncodedValue subnetworkEnc) {
         // partition graph into strongly connected components using Tarjan's algorithm
         StopWatch sw = new StopWatch().start();
-        EdgeBasedTarjanSCC.ConnectedComponents ccs = EdgeBasedTarjanSCC.findComponents(ghStorage, weighting, false);
+        EdgeBasedTarjanSCC.ConnectedComponents ccs = EdgeBasedTarjanSCC.findComponents(ghStorage,
+                (prev, edge) -> Double.isFinite(GHUtility.calcWeightWithTurnWeightWithAccess(weighting, edge, false, prev)),
+                false);
```

Comment:
Here we setup the EdgeTransitionFilter based on the weighting. We could go further and do this already in PrepareJob, but so far I do not see much benefit of this.